### PR TITLE
basic list view

### LINKED ISSUE #3: Display list view
Upon signing into the Lockbox app, I want to see a list of all my saved entries, by origin name/title. This is so I can easily find an entry, by scrolling, searching, or filtering, to get to that credential quickly.

## Acceptance criteria
* Should be able to view the list of entries by the origin
* Default alphabetical ordering on the origin should make sense (eg facebook comes before twitter)

### Visual Design

**InVision**
https://mozilla.invisionapp.com/share/3SN35UHWJDX#/309979873_Entry_List

**Zeplin**
https://zpl.io/b6MPKgy

### LINKED ISSUE #4: Provide entry titles in the app to make entries more human readable (entry list)
When viewing the entry list, I want to quickly scan the entry list by title (e.g., Facebook instead of https://www.facebook.com or even www.facebook.com), so I can quickly access my username and password for logging in. 

Problem: No Entry Name (title) currently available in Saved Logins so we need to derive one

## Acceptance criteria
* Should be able to view the list of entries by the origin
* Default alphabetical ordering on the origin should make sense (eg facebook comes before twitter)
* Should not not include the protocol (http://, https://, www.)
* Should not see the suffix or anything after the suffix (.com/{*})
* Should support future goals of editing the entry name

### Visual Design

**InVision**
https://mozilla.invisionapp.com/share/3SN35UHWJDX#/309979873_Entry_List

**Zeplin**
https://zpl.io/b6MPKgy

### PR DESCRIPTION
**Pull request by sashei**
resolves #3 
resolves #4

note that while all _view_ elements are present (i.e., search icon and title), they do not actually do anything yet.

### CODE REVIEW

**Comment by linuxwolf** [comment]
Running this branch in my simulators, the drawer text and background are both white:
image

Is this expected for now?

**Comment by sashei** [comment]
@linuxwolf I think there were visual regressions in the welcome screen and the drawer; I'm letting them stand for now with the premise that visual tuning will happen in later tickets.

File: app/src/main/java/mozilla/lockbox/presenter/ItemListPresenter.kt [hunk 1]
```diff
@@ -10,36 +10,61 @@ import android.view.MenuItem
...
+                                username,
+                                it.id)
+                    }
                 }
-            }
-        }.addTo(compositeDisposable)
+                .subscribe(view::updateItems)
+                .addTo(compositeDisposable)
+
+        dispatcher.dispatch(DataStoreAction.Type.UNLOCK)
```
[the remark this answers is not in the document]
Comment (sashei) [reply app/src/main/java/mozilla/lockbox/presenter/ItemListPresenter.kt]: 🙏 word, I forgot the `TODO` to remove here. Thanks!

File: app/src/main/java/mozilla/lockbox/action/DataStoreAction.kt [hunk 2]
```diff
@@ -8,8 +8,8 @@ package mozilla.lockbox.action
 
 import mozilla.lockbox.flux.Action
 
-data class DataStoreAction(val type: Type) : Action {
-    enum class Type {
+data class DataStoreAction(val type: Type) {
```
[the remark this answers is not in the document]
Comment (sashei) [reply app/src/main/java/mozilla/lockbox/action/DataStoreAction.kt]: 👍 I was a little confused when implementing initially. I don't love typing `DataStoreAction(DataStoreAction.Type...)` all the time but I don't think the change belongs here.

**Comment by sashei** [comment]
✅ 🎉 double review 🎉 ✅

**Comment by nickbrandt** [comment]
@sashei the text sizes for the list items still seems a bit small. I'm a little confused, as looking at the code I see you're using 16sp for the Title, and 14sp for the Sub-title - which is correct. When I compare this to the designs though, it shows up smaller. If I increase the default text size on my device by 1, it appears correct. The same is the case for the navbar title. Any idea why the sizing would not match at the default size?

screen shot 2018-09-26 at 9 26 17 am

File: app/src/main/res/values/styles.xml [hunk 3]
```diff
@@ -7,11 +7,26 @@
 <resources>
 
     <!-- Base application theme. -->
-    <style name="AppTheme" parent="Theme.AppCompat.Light.DarkActionBar">
+    <style name="AppTheme" parent="Theme.AppCompat.Light.NoActionBar">
         <!-- Customize your theme here. -->
         <item name="colorPrimary">@color/colorPrimary</item>
         <item name="colorPrimaryDark">@color/colorPrimaryDark</item>
-        <item name="colorAccent">@color/colorAccent</item>
+        <item name="android:textColor">#ffffff</item>
```
Comment (jhugman) [inline app/src/main/res/values/styles.xml]: Nit: this is a placeholder; can we make it at least legible? :)

**Pull request merged**